common/src/main/java/xyz/wagyourtail/jsmacros/client/util/NameUtil.java

```diff
@@ -0,0 +1,122 @@
+package xyz.wagyourtail.jsmacros.client.util;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Etheradon
+ * @since 1.8.4
+ */
+public final class NameUtil {
+
+    private static final String VALID_NAME = "[A-Za-z0-9_]+";
+    private static final Pattern PATTERN_NAME = Pattern.compile(VALID_NAME);
+    private static final Pattern PATTERN_WHISPER = Pattern.compile("(" + VALID_NAME + ") whispers to you.*");
+
+    private static final int MAX_STRING_LENGTH = 60;
+    private static final int DEFAULT_BUILDER_CAPACITY = 16;
+
+    private NameUtil() {
+    }
+
+    /**
+     * An approach to retrieve the name of a player from a chat message.
+     * <p>Iterate until <. If found, get the next valid name
+     * <p>Iterate until :. If found, get the last valid name
+     * <p>Iterate until >>. If found get the last valid name
+     * <p>Iterate until ->. If found get the last valid name (continue if empty)
+     * <p>The potential last name is the last value between [] or before >. It's used when the
+     * sentence just continues.
+     * <p>There are some edge cases like "name > <WrongName>", because it could also be "Guild >
+     * name :". Since the last one is much more common, I will stay with this approach.
+     *
+     * @return a list of the name at index 0, will be empty if no name was found and all titles,
+     *         which were inside [] or (), as the arguments
+     */
+    public static List<String> guessNameAndRoles(String text) {
```

Comment:
where is this used?... nowhere important I hope

Reply:
It's used in the Util Library. Many people were trying to get the author of chat messages, which was prior to 1.19 (and with no chat reports or similiar mods / plugins still is) only possible with customized regex for specific servers, due to them having different message formats. That's why I wrote this algorithm to extract the name, and because of the inner workings of the algorithm also the roles, of any given chat message. For this I looked at the most used plugins and servers to support nearly every server and message format.

Reply:
ok.maybe put a note in utils that it's a not guarentee'd to work and regex parsing may work better in some cases

Reply:
It already states `Tries to guess the name of the sender of a given message`, but I will add the information that regex may be better for specific servers or pruposes.